### ISSUE TITLE
Assets should not be loaded until entity and component is not enabled

### ISSUE THREAD

**Issue by PsichiX** [playcanvas/engine#442]
Hi, guys!
I've found a bug, that causes loading assets when they are not needed.

**Look at this scene and check Network report with listed resources loaded:**
http://playcanvas.com/editor/scene/394000/launch

**Now let me explain what's happening:**
I have scene with some entity that is disabled on start, so the scene is basically empty - nothing is renderred. So logicaly: there should not be loaded any assets of that disabled entity, but they are loaded. When you scale that problem for a very big game, then there are alot of assets that are not loaded on-demand but on game start (the difference between that and preloading stage is that preloading is done asynchronously, not before scene is initialized) - that's definitely a bug.

**How it should work:**
Imageine that you've multiple stages (or dialogs - the same idea of separated temporary parts of scene) on scene as disabled entities parents for given stages content. Now you'll want to switch to another stage so you'll disable current stage entity (then perform removing assets loaded for current stage) and next enable another stage and at this moment game should load assets used by new stage. The difference is that assets should be loaded only, and only when they're really used, not just assigned to entities.
Long story short: right now assets are loaded no matter if they're currently used or not (they're loaded when they're assigned to some entity component, no matter if that entity/component is enabled or not). To make it properly assets should be loaded not by assigning to some object, but by assigning into enabled entity/component or when they're assigned into disabled entity/component and entity/component is enabled.

Cheers! ^^

![2015-11-18 22_18_26-launch](https://cloud.githubusercontent.com/assets/2111867/11254769/d9074626-8e42-11e5-813b-24211a5dbde1.jpg)

**Comment by Maksims** [comment]
Hi. I've noticed this when was doing some demo quiet a long time ago. And do agree that if asset is set to non-preloaded, it should load ONLY if it is enabled and is in scene.

**Comment by vkalpias** [comment]
Fixed in https://github.com/playcanvas/engine/commit/89aaa6f65795269cf90c26bc4e64cda1282eca9f
